auto-run-dockerized-nondex/entrypoint.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+set -e
+
+# Checking if the git repo url is set
+if [ -z "$REPO_URL" ]; then
+  echo "Error: REPO_URL is not set. It is a required argument."
+  exit 1
+fi
+
+echo "Cloning repository: $REPO_URL"
+git clone "$REPO_URL" repo
+
+cd repo
+
+echo "Printing java version for reference"
+java --version
+
+# Initialize the Maven install command
+MVN_INSTALL_CMD="mvn clean install"
+
+# Add the -pl option if MODULE is provided
+if [ -n "$MODULE" ]; then
+  MVN_INSTALL_CMD="$MVN_INSTALL_CMD -pl $MODULE"
+fi
+
+# Output and execute the command
+echo "Running 'mvn clean install' with command: $MVN_INSTALL_CMD"
+eval $MVN_INSTALL_CMD
+
+# Creating a directory for logs, volume mount to get onto your local
+NONDEX_LOG_DIR=/app/nondex_logs
+mkdir -p "$NONDEX_LOG_DIR"
+
+# Building nondex command using the various options configured - eg. mvn -pl <module_path> edu.illinois:nondex-maven-plugin:2.1.7:nondex -Dtest="<tests>" --fail-never
+NONDEX_CMD="mvn"
+
+if [ -n "$MODULE" ]; then
+  NONDEX_CMD="$NONDEX_CMD -pl $MODULE"
+fi
+
+NONDEX_CMD="$NONDEX_CMD edu.illinois:nondex-maven-plugin:2.1.7:nondex"
+
+if [ -n "$TESTS" ]; then
+  NONDEX_CMD="$NONDEX_CMD -Dtest=\"$TESTS\""
+fi
+
+if [ -n "$NONDEX_RUNS" ]; then
+  NONDEX_CMD="$NONDEX_CMD -DnondexRuns=$NONDEX_RUNS"
+fi
+
+if [ "$RUN_NONDEX_WITH_FN" == "true" ]; then
+  NONDEX_CMD="$NONDEX_CMD --fail-never"
+fi
+
+echo "Final NonDex command: $NONDEX_CMD"
+
+NONDEX_LOG_FILE="$NONDEX_LOG_DIR/nondex-logs-$(date +%Y%m%d-%H%M%S).log"
+NONDEX_FAILURE_FILE="$NONDEX_LOG_DIR/nondex-flaky-tests-$(date +%Y%m%d-%H%M%S).txt"
+
+echo "Running the NonDex plugin. Logs will be saved to $NONDEX_LOG_FILE"
+eval $NONDEX_CMD | tee "$NONDEX_LOG_FILE"
```

Comment:
`eval $NONDEX_CMD 2>&1 | tee ...` to capture both stdout and stderr

Reply:
Makes sense. I agree with this. I'll make the change.